src/commands/app/downloadTemplates.ts

```diff
@@ -0,0 +1,89 @@
+import fs from "fs-extra";
+import { mkdir as mkdirCB } from "fs";
+import { promisify } from "util";
+
+import inquirer from "inquirer";
+import fetch from "node-fetch";
+
+import { Arguments, createCommand } from "../../util";
+import { requireApp, requireClusterConfig, requireUser } from "../middleware";
+import { cliContainer } from "../../container";
+import {
+  templateItemToLocalTemplatePath,
+  printTemplateItems,
+} from "./templateHelper";
+
+const mkdir = promisify(mkdirCB);
+
+async function run(argv: Arguments) {
+  const templateDir = argv.dir as string;
+  const remoteTemplates = await cliContainer.getTemplates(
+    argv.context.app || ""
+  );
+
+  const overwrites = [];
+  for (const i of remoteTemplates) {
+    const p = templateItemToLocalTemplatePath(i, templateDir);
+    try {
+      await fs.access(p);
+      overwrites.push(i);
+    } catch (e) {}
+  }
+
+  if (overwrites.length > 0) {
+    printTemplateItems(
+      overwrites,
+      "The following templates will be overwritten:",
+      templateDir
+    );
+    if (!argv.yes) {
+      const answers = await inquirer.prompt([
+        {
+          message: "Continue?",
+          name: "confirm",
+          type: "confirm",
+        },
+      ]);
+      if (!answers.confirm) {
+        return;
+      }
+    }
+  }
+
+  await mkdir(templateDir, { recursive: true });
+
+  for (const i of remoteTemplates) {
+    const p = templateItemToLocalTemplatePath(i, templateDir);
+    const resp = await fetch(i.url);
+    await new Promise((resolve, reject) => {
```

Comment:
We should do it as atomic operation; it's better to read entire response before writing it on disk.